src/tx/builder/helpers.ts

```diff
@@ -57,13 +57,16 @@ export function buildContractId (ownerId: string, nonce: number): string {
  * @param oracleId The oracle public key
  * @return Contract public key
  */
-export function oracleQueryId (senderId: string, nonce: number, oracleId: string): string {
+export function oracleQueryId (
+  senderId: string,
+  nonce: number | BigNumber,
+  oracleId: string): string {
   function _int32 (val: number): Buffer {
     const nonceBE = toBytes(val, true)
     return Buffer.concat([Buffer.alloc(32 - nonceBE.length), nonceBE])
   }
 
-  const b2bHash = hash(Buffer.from([...decode(senderId as EncodedData<'ak'>, 'ak'), ..._int32(nonce), ...decode(oracleId as EncodedData<'ok'>, 'ok')]))
+  const b2bHash = hash(Buffer.from([...decode(senderId as EncodedData<'ak'>, 'ak'), ..._int32(nonce as number), ...decode(oracleId as EncodedData<'ok'>, 'ok')]))
```

Comment:
```suggestion
  const b2bHash = hash(Buffer.from([...decode(senderId as EncodedData<'ak'>, 'ak'), ..._int32(nonce as number), ...decode(oracleId, 'ok')]))
```
only if you also accept the previous suggestion